### PR TITLE
perf: apply the final logit softcap on the GPU instead of a CPU loop

### PR DESCRIPTION
**Pull request by ericcurtin**
## What

`forward_gpu` applied the final logit softcap (`(logits/cap).tanh()*cap`) with a single-threaded CPU loop over the entire vocab (262144 elements for Gemma4-E2B) after the LM head's matvec already ran on the GPU. Measured in isolation, that loop takes **~1.1ms every decode step**, regardless of GPU availability.

This extends the LM head's existing `vkQueueSubmit` with a broadcast-mul(1/cap) -> tanh -> broadcast-mul(cap) dispatch chain instead, reusing the same broadcast-scalar trick already used for the per-layer `layer_scalar` multiply and the already-compiled-but-unused `tanh_f32` shader.

While rewriting this, I also removed the ad-hoc "find or lazily allocate a buffer of this exact byte size" logic the old LM head code used — replaced by an explicit `ACT_LOGIT_RAW` slot allocated up front like every other activation buffer in this file. This is both simpler and removes a latent bug: with four same-sized `[vocab]`-length buffers now needed (raw/scaled/tanh/final), "find the first buffer matching this size" would have started returning the wrong buffer.

## Verification

**Correctness**: `softcap_tests::gpu_softcap_vs_cpu_softcap` runs both the CPU loop and the new GPU dispatch chain against the same synthetic logits (at the real `vocab_size=262144` scale — this is a `[vocab]`-sized elementwise op, not a `[vocab, hidden]` weight matrix, so it's a ~1MB buffer, nowhere near the GPU-allocation limits a much larger synthetic matvec weight hit in an earlier investigation) and diffs every element: max abs err `0.000008`, negligible.

**Performance**: same test measures the CPU loop against the GPU chain directly:

```
softcap over 262144 elements: CPU 1143.5-1144.5us/call   GPU (3 dispatches, 1 submit) 521.0-542.5us/call   speedup 2.11-2.20x
```

Consistent across repeated runs. Unlike `input_layernorm`'s GPU move (#29 — measurably equal, not faster, since that op was tiny relative to the submit it joined), this op is large enough (262144 elements) that even with three extra dispatches' overhead, the GPU chain wins outright.

`cargo build --release`, `cargo test --release` (all 8 tests, including the 6 pre-existing ones exercising the layer stack, still pass at their established tolerances), and `cargo clippy --release --all-targets` are all clean — and one *fewer* clippy warning than `main`.

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -1123,66 +1140,65 @@ impl VulkanModel {
...
+        // ACT_LOGIT_RAW's doc comment). Measured ~2.1-2.2x faster than the
+        // CPU loop in isolation (softcap_tests below).
         let vocab = cfg.vocab_size;
-        let cap = cfg.final_logit_softcapping;
-        let mut logits = if let (Some(eng), Some(lm_w_ptr)) = (
+        let logits = if let (Some(eng), Some(lm_w_ptr)) = (
             self.engine.as_mut(),
             self.gpu_weights.get("model.embed_tokens.weight")
                 .map(|b| b as *const compute::Buffer)
         ) {
```
[the remark this answers is not in the document]
Comment (ericcurtin) [reply src/lib.rs]: Fixed in 39e4193 — added a `use_gpu_lm_head` gate requiring `self.init_act_bufs()` to actually succeed (mirroring forward_layer_gpu_matmuls's own use_gpu/use_fused_qkv/use_fused_post_attn pattern) before entering the GPU branch, falling back to the CPU path otherwise. Good catch — this risk pre-existed the change in this PR (the original code's act_ptr_mut(ACT_QKV_IN) had the same unconditional-index issue), but worth fixing now.
